lib/game.rb

```diff
@@ -0,0 +1,96 @@
+require 'yaml'
+require_relative 'validate'
+require_relative 'show_content'
+require_relative 'db_methods'
+require_relative 'user'
+module NewSuperCodebreaker2021
+  class Game
+    include Validate
+    include ShowContent
+    include DBMethods
+
+    def initialize
+      @code = generate_code
+    end
+
+    GUESS_COMMANDS = %i[hint rules exit].freeze
+    START_COMMANDS = %i[start rules stats exit].freeze
+    DIFFICULTY_COMMANDS = %i[easy medium hell exit].freeze
+    AFTER_GAME_COMMANDS = %i[start save exit].freeze
+    YES_NO_COMMANDS = %i[yes no].freeze
+
+    def chose_command(command)
+      check_input(command, START_COMMANDS)
+    end
+
+    def take_name(input_name)
+      if input_name == 'exit'
+        :exit
+      else
+        validate_name(input_name)
+      end
+    end
+
+    def chose_difficulty(difficulty)
+      check_input(difficulty, DIFFICULTY_COMMANDS)
+    end
+
+    def user_guess(code)
+      if code.to_i != 0
+        validate_user_code(code)
+      elsif GUESS_COMMANDS.include?(code.to_sym)
+        code.to_sym
+      else false
+      end
+    end
+
+    def take_hint(user, used_hints)
+      code_copy = @code.dup
```

Comment:
removed @code.dup  to initialize

Reply:
if @code has 1222, take_hint will return [1, 2] because the assignment says that numbers can't be repeated:
If I take a hint 2 or more times - I must receive back different digits of a secret number, so I mustn’t receive the same digit number.